### PR TITLE
CLDC-2274 Fix buyer 2 nationality routing bug

### PR DESCRIPTION
**Pull request by natdeanlewissoftwire**
ticket: https://digital.dclg.gov.uk/jira/browse/CLDC-2274

Buyer 2 nationality was displaying regardless of buyer being interviewed or not - it should follow the same pattern as every other question in the subsection, in that it should only display when the bueyr has seen the privacy notice or was not interviewed. This is for the same reason as every other question in this subsection.

fixed behaviour: (previously buyer 2 nationality was displaying at the bottom of this page in this case)
image

### CODE REVIEW

File: app/models/form/sales/pages/buyer2_nationality.rb [hunk 1]
```diff
@@ -2,7 +2,16 @@ class Form::Sales::Pages::Buyer2Nationality < ::Form::Page
   def initialize(id, hsh, subsection)
     super
     @id = "buyer_2_nationality"
-    @depends_on = [{ "joint_purchase?" => true }]
+    @depends_on = [
```
Comment (natdeanlewissoftwire) [inline app/models/form/sales/pages/buyer2_nationality.rb:5]: this contains the actual behaviour for the ticket, rest is just refactoring

**Comment by jamesrose** [comment]
@natdeanlewissoftwire Please could you include what was fixed and why?

**Comment by natdeanlewissoftwire** [comment]
> @natdeanlewissoftwire Please could you include what was fixed and why?

@jamesrose done 👍

File: app/models/form/sales/pages/housing_benefits.rb [hunk 2]
```diff
@@ -2,7 +2,14 @@ class Form::Sales::Pages::HousingBenefits < ::Form::Page
   def initialize(id, hsh, subsection, joint_purchase:)
     super(id, hsh, subsection)
     @joint_purchase = joint_purchase
-    @depends_on = [{ "jointpur" => @joint_purchase ? 1 : 2 }]
+    @depends_on =
```
Comment (asmega) [inline app/models/form/sales/pages/housing_benefits.rb]: it works but think i would extract to a method at this point. not sure how i feel having an `if` within an array?

File: db/schema.rb [hunk 3]
```diff
@@ -519,6 +519,7 @@
     t.integer "prevten"
     t.integer "mortgageused"
     t.integer "wchair"
+    t.integer "income2_value_check"
```
Comment (asmega) [inline db/schema.rb]: i don't recommend checking in the schema file if you haven't actually made any schema changes

**Review by asmega — APPROVED** [review]

File: db/schema.rb (hunk 3 above)
Comment (natdeanlewissoftwire) [reply db/schema.rb]: good point

File: app/models/form/sales/pages/housing_benefits.rb (hunk 2 above)
Comment (natdeanlewissoftwire) [reply app/models/form/sales/pages/housing_benefits.rb]: yep agreed

**Pull request merged**